plugins/jobs/src/js/JobDelete.js

```diff
@@ -0,0 +1,90 @@
+import * as React from "react";
+import { componentFromStream, graphqlObservable } from "data-service";
+import { Subject } from "rxjs/Subject";
+import "rxjs/add/operator/map";
+import "rxjs/add/operator/switchMap";
+import "rxjs/add/operator/do";
+import "rxjs/add/operator/combineLatest";
+import "rxjs/add/operator/startWith";
+import "rxjs/add/operator/catch";
+
+import gql from "graphql-tag";
+
+import JobDeleteModal from "./components/JobDeleteModal";
+import defaultSchema from "./data/JobModel";
+
+const deleteJob = gql`
+  mutation {
+    deleteJob(id: $jobId, stopCurrentJobRuns: $stopCurrentJobRuns) {
+      jobId
+    }
+  }
+`;
+
+function executeDelete({ jobId, stopCurrentJobRuns, onSuccess, errorMsg }) {
+  return graphqlObservable(deleteJob, defaultSchema, {
+    jobId,
+    stopCurrentJobRuns
+  })
+    .map(_ => ({ done: true, stopCurrentJobRuns, errorMsg }))
+    .do(_ => onSuccess())
+    .startWith({ done: false, stopCurrentJobRuns, errorMsg });
+}
+
+function deleteOperation() {
+  const deleteSubject$ = new Subject();
+  const delete$ = deleteSubject$.switchMap(executeDelete).catch(error => {
+    return delete$.startWith({
+      errorMsg: error.response.message,
+      done: true
+    });
+  });
+
+  return {
+    delete$,
+    deleteHandler: (jobId, stopCurrentJobRuns, onSuccess, errorMsg) => {
+      deleteSubject$.next({ jobId, stopCurrentJobRuns, onSuccess, errorMsg });
+    }
+  };
+}
+
+function errorIsTaskCurrentRunning(errorMsg) {
```

Comment:
I like the verbosity of this funcion name a lot :)

Reply:
It's hard to understand what's going on from the name. Will it error out? Will it detect error? Shall it be `errorIfTaskCurrentRunning` or `errorIfCurrentTaskIsRunning`?

Reply:
What about: isTaskCurrentRunning?

Reply:
Is `TaskCurrent` an entity? Why then not `CurrentTask`?